### PR TITLE
Switch to a federated serverlite extension

### PR DESCRIPTION
**Pull request by jtpio**
Continue the switch to a federated serverlite extension:

- https://github.com/jupyterlite/jupyterlite/pull/104
- https://github.com/jupyterlite/jupyterlite/pull/352

### TODO

- [x] Switch back to `0.1.0-alpha.x` versioning
- [x] Add `liteExtension` key to the `package.json` to indicate this is not a regular lab extension
- [x] Fix CI
- [x] Fix releaser config
- [x] Add docs to the readme
- [x] Rename Python package to `jupyterlite-p5-kernel`?
- [x] Book name on `PyPI`: https://pypi.org/project/jupyterlite-p5-kernel/

### CODE REVIEW

**Comment by bollwyvl** [comment]
Just thinking out loud: the happy path might be:
- `labextension build`
  - puts stuff in `p5_kernel/labextension` 
- `setup.py sdist bdist_wheel`
  - puts stuff in `$CRAZYTOWN/share/jupyter/labextensions`
- `pip install jupyterlite-p5-kernel`
  - puts stuff in `$PREFIX/share/jupyter/labextensions`
- `jupyter lite build`
  - finds it
    - add to `federated_extensions` 
    - looks in `package.json`
      - find it's a serverlite_extension
        - add to `lite_extensions`
- at run-time
  - hand to the `JupyterBackEndApplication`
  - _don't_ hand to the `JupyterFrontentEndApplication`

easy, right!?

File: p5_kernel_lite/_version.py [hunk 1]
```diff
@@ -3,16 +3,23 @@
...
     HERE = Path(__file__).parent.resolve()
 
-    for settings in HERE.rglob("package.json"): 
+    for settings in HERE.rglob("package.json"):
         try:
             with settings.open() as f:
-                return json.load(f)["version"]
+                version = json.load(f)["version"]
+                return (
+                    version.replace("-alpha.", "a")
```
Comment (bollwyvl) [inline p5_kernel_lite/_version.py:15]: blearg, how many times have i written this :rofl:

**Comment by bollwyvl** [comment]
> `import { JavaScriptKernel } from '@jupyterlite/javascript-kernel'`

Should we also `sharedPackage` for `@jupyterlite/javascript-kernel`, at least for the time being?

**Comment by jtpio** [comment]
> Should we also `sharedPackage` for `@jupyterlite/javascript-kernel`, at least for the time being?

Yes sounds good :+1: At least if we continue shipping it by default. Otherwise it would be using its own copy of `@jupyterlite/javascript-kernel` which should also be fine for now.

**Comment by jtpio** [comment]
> Just thinking out loud: the happy path might be:

Yes, which should almost correspond to the current state of https://github.com/jupyterlite/jupyterlite/pull/352.

I'll finish this PR and cut a first release of `p5-kernel-lite` to PyPI so we can test on https://github.com/jupyterlite/jupyterlite/pull/352

For the name of the federated kernel, we could go for:

- `jupyterlite-p5-kernel`
- `p5-kernel-lite`

The second sounds good, but would be fine with the first one too. Later we can imagine having `xeus-lua-lite` which might look better than `jupyterlite-xeus-lua`.

**Comment by jtpio** [comment]
> The second sounds good, but would be fine with the first one too

`jupyterlite-p5-kernel` is actually similar to what many lab extensions have been using (`jupyterlab-myextension`). So going with that name would also be fine.

**Comment by jtpio** [comment]
Should be good to go for a first iteration. We can iterate in follow-up PRs.

**Pull request merged**